pyccel/ast/bind_c.py

```diff
@@ -312,94 +312,49 @@ def wrapping_bound_argument(self):
 
 # =======================================================================================
 
-
-class BindCFunctionDefResult(FunctionDefResult):
+class BindCResultVariable(Variable):
```

Comment:
Maybe this new class could be useful outside the concept of results in the future. Can we find a better name? `BindCVariable` is already taken (for module variables). Maybe the clearest name involves renaming `BindCVariable`?

Reply:
I agree! maybe we should rename `BindCVariable`

Reply:
`BindCResultVariable`->`BindCVariable`
`BindCVariable`->`BindCModuleVariable`
?

Reply:
Done:
[5520a73](https://github.com/pyccel/pyccel/pull/2157/commits/5520a7341268bc0eaa8e973f07f612b45dd94d0d)
[e940783](https://github.com/pyccel/pyccel/pull/2157/commits/e940783c0fc887fc5bfdaf3ba7331f62f58a5f64)
[9f258dd](https://github.com/pyccel/pyccel/pull/2157/commits/9f258dda14ac986655016134c974813178d8acdc)
[9161b46](https://github.com/pyccel/pyccel/pull/2157/commits/9161b46205cacf5e62eb5f8d2bd2c1899f951a5e)